source/math/FloatBox2D.ooc

```diff
@@ -111,6 +111,20 @@ FloatBox2D: cover {
 				result add(i)
 		result
 	}
+	separatedDistance: func (point: FloatPoint2D) -> FloatPoint2D {
+		distanceX := Float absolute(point x - this center x) - this width / 2.f
+		distanceY := Float absolute(point y - this center y) - this height / 2.f
+		FloatPoint2D new(distanceX, distanceY) maximum(FloatPoint2D new(0.f, 0.f))
```

Comment:
If you add an `absolute` property to `FloatSize2D`
```ooc
absolute ::= This new(Float absolute(this width), Float absolute(this height))
```
, these three lines can be reduced to
```ooc
((point - this center) absolute - this size / 2.f) maximum(FloatSize new())
```